### PR TITLE
[DataObject] Added mandatory check for ExternalImage data type

### PR DESCRIPTION
**Pull request by MyZik**
## Changes in this pull request  
Actually if we have a mandatory field with `ExternalImage` data type, the mandatory check will be skipped:

Bildschirmfoto 2021-09-07 um 12 21 27

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/ExternalImage.php [hunk 1]
```diff
@@ -298,6 +298,22 @@ public function synchronizeWithMasterDefinition(Model\DataObject\ClassDefinition
...
+     *
+     * @throws Model\Element\ValidationException
+     */
+    public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
+    {
+        if ($this->getMandatory() && !$omitMandatoryCheck && !$this->isEmpty($data)) {
+            throw new Model\Element\ValidationException('[ ' . $this->getName() . " ] should not be empty!");
+        }
+
+        parent::checkValidity($data, $omitMandatoryCheck);
```
Comment (brusch) [inline models/DataObject/ClassDefinition/Data/ExternalImage.php]:
```suggestion
```
Don't think we actually need that.

File: models/DataObject/ClassDefinition/Data/ExternalImage.php [hunk 2]
```diff
@@ -298,6 +298,21 @@ public function synchronizeWithMasterDefinition(Model\DataObject\ClassDefinition
...
+     * @param DataObject\Data\ExternalImage|null $data
+     * @param bool $omitMandatoryCheck
+     * @param array $params
+     *
+     * @throws Model\Element\ValidationException
+     */
+    public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
+    {
+        if ($this->getMandatory() && !$omitMandatoryCheck && !$this->isEmpty($data)) {
+            throw new Model\Element\ValidationException('[ ' . $this->getName() . " ] should not be empty!");
```
Comment (brusch) [inline models/DataObject/ClassDefinition/Data/ExternalImage.php]:
```suggestion
            throw new Model\Element\ValidationException('Empty mandatory field [ ' . $this->getName() . ' ]');
```
let's just go for the message from the parent, to be consistent 😉

**Comment by brusch** [comment]
@MyZik I'd classify this as a bug, could you please rebase this PR onto `10.1`? Thanks!

**Comment by MyZik** [comment]
@brusch I have made the rebase, is it correct? A lot of commits...

```
git remote add upstream https://github.com/pimcore/pimcore.git
git rebase --onto upstream/10.1 master
git push origin fix/image-advanced-omit-check -f
```

**Comment by brusch** [comment]
Unfortunately doesn't look right. If you like you can just close this PR and re-open another one 😉

**Comment by MyZik** [comment]
Okay, 1 min.

**Pull request closed without merging**